cmd/perf-tests.go

```diff
@@ -338,3 +340,80 @@ func netperf(ctx context.Context, duration time.Duration) madmin.NetperfNodeResu
 	globalNetPerfRX.Reset()
 	return madmin.NetperfNodeResult{Endpoint: "", TX: r.n / uint64(duration.Seconds()), RX: uint64(rx / delta.Seconds()), Error: errStr}
 }
+
+func siteNetperf(ctx context.Context, duration time.Duration) madmin.SiteNetPerfNodeResult {
+	r := &netperfReader{eof: make(chan struct{})}
+	r.buf = make([]byte, 128*humanize.KiByte)
+	rand.Read(r.buf)
+
+	connectionsPerPeer := 16
+
+	clusterInfos, err := globalSiteReplicationSys.GetClusterInfo(ctx)
+	if err != nil {
+		return madmin.SiteNetPerfNodeResult{}
+	}
+
+	if len(clusterInfos.Sites) > 16 {
+		// For a large cluster it's enough to have 1 connection per peer to saturate the network.
+		connectionsPerPeer = 1
+	}
+
+	errStr := ""
+	var wg sync.WaitGroup
+
+	for _, info := range clusterInfos.Sites {
+		// skip self
+		if globalDeploymentID == info.DeploymentID {
+			continue
+		}
+		info := info
+		for i := 0; i < connectionsPerPeer; i++ {
+			wg.Add(1)
+			go func() {
```

Comment:
What does this actually do?

I don't see any references to `r` here. This does *one* remote call?

Reply:
```
				}
				req, err := http.NewRequestWithContext(ctx, http.MethodPost, reqURL.String(), r)
				if err != nil {
					return
				}
				client := &http.Client{
					Timeout:   duration + 10*time.Second,
					Transport: globalRemoteTargetTransport,
				}
```
here